resolwe/flow/management/commands/list_docker_images.py

```diff
@@ -22,6 +23,11 @@
 
 logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
 
+# Global list of images already pulled when running in the same process. This is
+# used to avoid pulling images more than once when executed from tests.
+PULLED_IMAGES = set()
+PULLED_IMAGES_LOCK = threading.Lock()
+
```

Comment:
Could you extend the functionality to also avoid pulling images that failed more than once?

On my example Resolwe test suite run, the invalid `resolwe/test:versioning-2` Docker image tried to be pulled 8 times:

```
Trying to pull repository docker.io/resolwe/test ... 
sha256:7dfffa13a2addc317ac3bdfbddbd4604ea629decea19c271481e5c45245b7612: Pulling from docker.io/resolwe/test
Digest: sha256:7dfffa13a2addc317ac3bdfbddbd4604ea629decea19c271481e5c45245b7612
Status: Image is up to date for docker.io/resolwe/test:base
Docker image 'resolwe/test:base' pulled successfully!
Trying to pull repository docker.io/resolwe/test ... 
Trying to pull repository registry.fedoraproject.org/resolwe/test ... 
Trying to pull repository registry.access.redhat.com/resolwe/test ... 
Trying to pull repository docker.io/resolwe/test ... 
manifest for docker.io/resolwe/test:versioning-2 not found
Failed to pull Docker image 'resolwe/test:versioning-2'!
Trying to pull repository docker.io/resolwe/test ... 
sha256:e4b06e7197bec5e8be1d14dd12ad6a56c231771919ae8b6c6dea61f88533081e: Pulling from docker.io/resolwe/test
Digest: sha256:e4b06e7197bec5e8be1d14dd12ad6a56c231771919ae8b6c6dea61f88533081e
Status: Image is up to date for docker.io/resolwe/test:requirements-docker
Docker image 'resolwe/test:requirements-docker' pulled successfully!
ok
test_docker_uid_gid (resolwe.flow.tests.test_executors.ManagerRunProcessTest) ... 
Trying to pull repository docker.io/resolwe/test ... 
Trying to pull repository registry.fedoraproject.org/resolwe/test ... 
Trying to pull repository registry.access.redhat.com/resolwe/test ... 
Trying to pull repository docker.io/resolwe/test ... 
manifest for docker.io/resolwe/test:versioning-2 not found
Failed to pull Docker image 'resolwe/test:versioning-2'!
ok
test_executor_fs_lock (resolwe.flow.tests.test_executors.ManagerRunProcessTest) ... 
Trying to pull repository docker.io/resolwe/test ... 
Trying to pull repository registry.fedoraproject.org/resolwe/test ... 
Trying to pull repository registry.access.redhat.com/resolwe/test ... 
Trying to pull repository docker.io/resolwe/test ... 
manifest for docker.io/resolwe/test:versioning-2 not found
Failed to pull Docker image 'resolwe/test:versioning-2'!
ok
test_executor_requirements (resolwe.flow.tests.test_executors.ManagerRunProcessTest) ... 
Trying to pull repository docker.io/resolwe/test ... 
Trying to pull repository registry.fedoraproject.org/resolwe/test ... 
Trying to pull repository registry.access.redhat.com/resolwe/test ... 
Trying to pull repository docker.io/resolwe/test ... 
manifest for docker.io/resolwe/test:versioning-2 not found
Failed to pull Docker image 'resolwe/test:versioning-2'!
ok
test_invalid_storage_file (resolwe.flow.tests.test_executors.ManagerRunProcessTest) ... ok
test_memory_resource (resolwe.flow.tests.test_executors.ManagerRunProcessTest) ... 
Trying to pull repository docker.io/resolwe/test ... 
Trying to pull repository registry.fedoraproject.org/resolwe/test ... 
Trying to pull repository registry.access.redhat.com/resolwe/test ... 
Trying to pull repository docker.io/resolwe/test ... 
manifest for docker.io/resolwe/test:versioning-2 not found
Failed to pull Docker image 'resolwe/test:versioning-2'!
ok
test_minimal_process (resolwe.flow.tests.test_executors.ManagerRunProcessTest) ... ok
test_missing_file (resolwe.flow.tests.test_executors.ManagerRunProcessTest) ... ok
test_network_resource (resolwe.flow.tests.test_executors.ManagerRunProcessTest) ... 
Trying to pull repository docker.io/resolwe/test ... 
Trying to pull repository registry.fedoraproject.org/resolwe/test ... 
Trying to pull repository registry.access.redhat.com/resolwe/test ... 
Trying to pull repository docker.io/resolwe/test ... 
manifest for docker.io/resolwe/test:versioning-2 not found
Failed to pull Docker image 'resolwe/test:versioning-2'!
ok
test_null_executor (resolwe.flow.tests.test_executors.ManagerRunProcessTest) ... ok
test_run_in_docker (resolwe.flow.tests.test_executors.ManagerRunProcessTest) ... 
Trying to pull repository docker.io/resolwe/test ... 
Trying to pull repository registry.fedoraproject.org/resolwe/test ... 
Trying to pull repository registry.access.redhat.com/resolwe/test ... 
Trying to pull repository docker.io/resolwe/test ... 
manifest for docker.io/resolwe/test:versioning-2 not found
Failed to pull Docker image 'resolwe/test:versioning-2'!
ok
test_scheduling_class (resolwe.flow.tests.test_executors.ManagerRunProcessTest) ... 
Trying to pull repository docker.io/resolwe/test ... 
Trying to pull repository registry.fedoraproject.org/resolwe/test ... 
Trying to pull repository registry.access.redhat.com/resolwe/test ... 
Trying to pull repository docker.io/resolwe/test ... 
manifest for docker.io/resolwe/test:versioning-2 not found
Failed to pull Docker image 'resolwe/test:versioning-2'!
```